static/app/views/dashboards/widgetBuilder/components/newWidgetBuilder.tsx

```diff
@@ -31,6 +54,45 @@ function WidgetBuilderV2({isOpen, onClose}: DevWidgetBuilderProps) {
           <WidgetBuilderProvider>
             <WidgetBuilderContainer>
               <WidgetBuilderSlideout isOpen={isOpen} onClose={onClose} />
+              <WidgetPreviewContainer
+                dashboardFilters={dashboardFilters}
+                dashboard={dashboard}
+              />
+            </WidgetBuilderContainer>
+          </WidgetBuilderProvider>
+        )}
+      </AnimatePresence>
+    </Fragment>
+  );
+}
+
+export default WidgetBuilderV2;
+
+function WidgetPreviewContainer({
+  dashboardFilters,
+  dashboard,
+}: {
+  dashboard: DashboardDetails;
+  dashboardFilters: DashboardFilters;
+}) {
+  const {state} = useWidgetBuilderContext();
+  const organization = useOrganization();
+  const location = useLocation();
+
+  return (
+    <DashboardsMEPProvider>
+      <MetricsCardinalityProvider organization={organization} location={location}>
+        <MetricsDataSwitcher
+          organization={organization}
+          location={location}
+          hideLoadingIndicator
+          eventView={EventView.fromLocation(location)}
```

Comment:
This is currently what's being passed down in the old widget builder, so I brought it over